File: week2/eunbeen/더맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> queue = new PriorityQueue();
        
        for(int rate : scoville){
            queue.add(rate);
        }
        
        while(queue.stream().anyMatch(e -> e < K) && queue.peek() < K && queue.size() >= 2){
            int first_food = queue.remove();
            int second_food = queue.remove();
            
            int mixed_food = first_food + (second_food * 2);
            queue.add(mixed_food);
            
            answer += 1;
        }
        
        if(queue.stream().anyMatch(e -> e < K)){
            return -1;
        }
        
        return answer;
    }
}

File: week2/cylee/더맵게.py
from queue import PriorityQueue
def solution(scoville, K): # 실패
    answer = 0
    
    que = PriorityQueue()
    for i in scoville:
        que.put(i)
    # 우선순위큐로 
    while que.qsize() > 1:
        first = que.get()
        second = que.get()
        total = first + (second*2)

        # 맨앞에 하나 꺼내본 다음에 판단
        que.put(total)
        temp = que.get()
        answer += 1
        if(temp >= K):
            break
        else:
            que.put(temp)
    return answer

print(solution([3, 2, 5, 9, 1, 12]	,7))